service/controller/app/v1/resource/chart/desired.go

```diff
@@ -0,0 +1,70 @@
+package chart
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	catalogName := key.CatalogName(cr)
+
+	appCatalog, err := r.g8sClient.ApplicationV1alpha1().AppCatalogs("default").Get(catalogName, v1.GetOptions{})
```

Comment:
"default" shouldn't be hardcoded. We should use the flag for the watch namespace.

https://github.com/giantswarm/app-operator/blob/aafeed93353391ff15f25ce33886f11f7d594371/flag/service/kubernetes/watch/watch.go#L6